### PR TITLE
Connectors for ADLS New OAuth2.0 methods and code cleanup

### PR DESCRIPTION
**Pull request by omkarksa**
This incorporates changes of the previous pull request (https://github.com/Microsoft/AzureSMR/pull/103), that adds new OAuth2.0 methods to the existing AzureSMR auth framework. Also addresses several review comments and extensive code cleanup.

### CODE REVIEW

**Comment by omkarksa** [comment]
@andrie Please "Squash and merge" these changes once you are done with your review.

**Comment by omkarksa** [comment]
@richcalaway Could you please help review the changes before merge?

**Comment by omkarksa** [comment]
@richcalaway Please let me know if you need any info regarding these changes.

**Comment by omkarksa** [comment]
@richcalaway I have addressed your CR comments and the pull request now includes the changes. Please check and let me know if is OK.

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
  
 ---
  
- `AzureSMR` is an R Package for managing a selection of Azure resources, using the Azure Service Manager API. The package exposes function to manage resources, resource groups, storage (blobs and containers), ARM templates, virtual machines and HDInsight (nodes, Hive and Spark). To use the package, you must configure an Azure Active Directory application and service principal in the Azure portal.
+ `AzureSMR` is an R Package for managing a selection of Azure resources, using the Azure Service Manager API. The package exposes function to manage resources, resource groups, storage (blobs and containers), ARM templates, virtual machines, HDInsight (nodes, Hive and Spark) and Azure Datalake Store. To use the package, you must configure an Azure Active Directory application and service principal in the Azure portal.
```
Comment (inchiosa) [inline README.md]: Please change to Azure Data Lake Store

File: vignettes/tutorial.Rmd [hunk 2]
```diff
@@ -59,6 +60,15 @@ sc <- createAzureContext(tenantID = "{TID}", clientID = "{CID}", authKey= "{KEY}
 sc
 ```
 
+or using the "DeviceCode" flow if supported by that resource by trying:
+
+```{r, eval = FALSE}
+sc <- createAzureContext(tenantID = "{TID}", clientID = "{CID}", authType= "DeviceCode")
+# Manually authenicate using DeviceCode flow
```
Comment (inchiosa) [inline vignettes/tutorial.Rmd]: spelling: authenticate

File: vignettes/tutorial.Rmd [hunk 3]
```diff
@@ -309,3 +319,51 @@ azureSparkCMD(sc, clustername = "smrhdi", CMD = "HW<-'hello R'", sessionID = "2"
...
 
+
+## Accessing Azure Data Lake Store using the `azureActiveContext`
+
+To access Azure Data Lake you need to generate an access token using either "ClientCredential" (default) or "DeviceCode" AuthType use `createAzureContext()`.
+```{r, eval=FALSE}
+asc <- createAzureContext(tenantID = "{TID}", clientID = "{CID}", authKey= "{KEY}")
+```
+
+To create directories in an azure datalake store account use `azureDataLakeMkdirs()`
```
Comment (inchiosa) [inline vignettes/tutorial.Rmd]: Please change to Azure Data Lake Store

File: vignettes/tutorial.Rmd [hunk 4]
```diff
@@ -309,3 +319,51 @@ azureSparkCMD(sc, clustername = "smrhdi", CMD = "HW<-'hello R'", sessionID = "2"
...
+```
+
+To list items in an azure datalake store account use `azureDataLakeListStatus()`
+
+```{r, eval=FALSE}
+azureDataLakeListStatus(asc, azureDataLakeAccount, "")
+azureDataLakeListStatus(asc, azureDataLakeAccount, "tempfolder")
+```
+
+To create a file and optionally write data to the new file in an azure data lake storage account use `azureDataLakeCreate()`
```
Comment (inchiosa) [inline vignettes/tutorial.Rmd]: Please change to Azure Data Lake Store

File: vignettes/tutorial.Rmd [hunk 5]
```diff
@@ -309,3 +319,51 @@ azureSparkCMD(sc, clustername = "smrhdi", CMD = "HW<-'hello R'", sessionID = "2"
...
+```
+
+To read a file in azure datalake store account use `azureDataLakeRead()`
+
+```{r, eval=FALSE}
+azureDataLakeRead(asc, azureDataLakeAccount, "tempfolder/tempfile00.txt", 
+                  length = 2L, bufferSize = 4194304L)
+```
+
+To delete item(s) in azure datalake store account use `azureDataLakeDelete()`
```
Comment (inchiosa) [inline vignettes/tutorial.Rmd]: Please change to Azure Data Lake Store

File: vignettes/tutorial.Rmd [hunk 6]
```diff
@@ -309,3 +319,51 @@ azureSparkCMD(sc, clustername = "smrhdi", CMD = "HW<-'hello R'", sessionID = "2"
...
+                    charToRaw("abcd"))
+```
+
+To append to a file in an azure data lake storage account use `azureDataLakeAppend()`
+
+```{r, eval=FALSE}
+azureDataLakeAppend(asc, azureDataLakeAccount, "tempfolder/tempfile00.txt", 4194304L, charToRaw("stuv"))
+```
+
+To read a file in azure datalake store account use `azureDataLakeRead()`
```
Comment (inchiosa) [inline vignettes/tutorial.Rmd]: Please change to Azure Data Lake Store

File: vignettes/tutorial.Rmd [hunk 7]
```diff
@@ -309,3 +319,51 @@ azureSparkCMD(sc, clustername = "smrhdi", CMD = "HW<-'hello R'", sessionID = "2"
...
+To create a file and optionally write data to the new file in an azure data lake storage account use `azureDataLakeCreate()`
+
+```{r, eval=FALSE}
+azureDataLakeCreate(asc, azureDataLakeAccount, "tempfolder/tempfile00.txt", 
+                    "755", FALSE, 
+                    4194304L, 3L, 268435456L, 
+                    charToRaw("abcd"))
+```
+
+To append to a file in an azure data lake storage account use `azureDataLakeAppend()`
```
Comment (inchiosa) [inline vignettes/tutorial.Rmd]: Please change to Azure Data Lake Store

File: vignettes/tutorial.Rmd [hunk 8]
```diff
@@ -309,3 +319,51 @@ azureSparkCMD(sc, clustername = "smrhdi", CMD = "HW<-'hello R'", sessionID = "2"
...
+asc <- createAzureContext(tenantID = "{TID}", clientID = "{CID}", authKey= "{KEY}")
+```
+
+To create directories in an azure datalake store account use `azureDataLakeMkdirs()`
+
+```{r, eval=FALSE}
+azureDataLakeMkdirs(asc, azureDataLakeAccount, "tempfolder")
+```
+
+To list items in an azure datalake store account use `azureDataLakeListStatus()`
```
Comment (inchiosa) [inline vignettes/tutorial.Rmd]: Please change to Azure Data Lake Store

File: README.md (hunk 1 above)
Comment (omkarksa) [reply README.md]: Done. Will update this PR.

File: vignettes/tutorial.Rmd (hunk 2 above)
Comment (omkarksa) [reply vignettes/tutorial.Rmd]: Done. Will update this PR.

**Comment by omkarksa** [comment]
Thanks @inchiosa. I have addressed your CR comments. Please check and let me know if its OK.

**Pull request merged**